File: ResourceManager/Models/Configurations/BookConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ResourceManager.Models.Configurations
{
    public class BookConfiguration : IEntityTypeConfiguration<Book>
    {
        public void Configure(EntityTypeBuilder<Book> builder)
        {
            builder.ToTable("Books");
            builder.HasKey(x => x.ID);

            builder.Property(x => x.ID).ValueGeneratedOnAdd();
            builder.Property(x => x.CreatedDate).HasDefaultValueSql("date('now','localtime')");
            builder.Property(x => x.ModifiedDate).HasDefaultValueSql("date('now','localtime')");
            builder.Property(x => x.Title).HasColumnType("nvarchar(255)").IsRequired();
            builder.Property(x => x.Author).HasColumnType("nvarchar(255)").IsRequired();
            builder.Property(x => x.Publisher).HasColumnType("nvarchar(100)").IsRequired();
            builder.Property(x => x.Filename).HasColumnType("nvarchar(255)").IsRequired();
            builder.Property(x => x.Cover).HasColumnType("nvarchar(255)").IsRequired(false);
            builder.Property(x => x.Abstraction).HasColumnType("nvarchar(255)").IsRequired(false);
            builder.HasIndex(x => x.Title).IsUnique();
        }
    }
}

File: ResourceManager/Models/Configurations/BookConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore
Imports System
Imports System.Linq

Namespace ResourceManager.Models.Configurations
    Public Class BookConfiguration
        Implements IEntityTypeConfiguration(Of Book)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of Book)) Implements IEntityTypeConfiguration(Of Book).Configure
            builder.ToTable("Books")
            builder.HasKey(Function(x) x.ID)

            builder.Property(Function(x) x.ID).ValueGeneratedOnAdd()
            builder.[Property](Function(x) x.CreatedDate).HasDefaultValueSql("date('now','localtime')")
            builder.[Property](Function(x) x.ModifiedDate).HasDefaultValueSql("date('now','localtime')")
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Title), "nvarchar(255)").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Author), "nvarchar(255)").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Publisher), "nvarchar(100)").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Filename), "nvarchar(255)").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Cover), "nvarchar(255)").IsRequired(False)
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Abstraction), "nvarchar(255)").IsRequired(False)
            builder.HasIndex(CType(Function(x) x.Title, Expressions.Expression(Of Func(Of Book, Object)))).IsUnique()
        End Sub
    End Class
End Namespace
